includes/Services/SiteGenService.php

```diff
@@ -284,6 +281,36 @@ public static function populate_fonts_in_theme_styles( $theme_styles, $homepage_
 		return $theme_styles;
 	}
 
+	/**
+	 * Updates the global theme for the sitegen flow.
+	 *
+	 * @param array $data Data on homepage and it's corresponding styles.
+	 * @return true|\WP_Error
+	 */
+	public static function update_styles_for_sitegen( $data ) {
+		global $wp_filesystem;
+
+		// Get the currently activated theme and the corresponding theme json
+		$active_theme_dir       = \get_template_directory();
+		$active_theme_json_file = $active_theme_dir . '/theme.json';
+
+		// If the theme json doesn't exist, give up
+		if ( ! $wp_filesystem->exists( $active_theme_json_file ) ) {
+			return false;
+		}
+
+		$active_theme_json      = $wp_filesystem->get_content( $active_theme_json_file );
+		$active_theme_json_data = json_decode( $active_theme_json_file, true );
```

Comment:
```suggestion
		$active_theme_json_data = json_decode( $active_theme_json, true );
```
Since `$active_theme_json_file` has only the filename.